assets/js/frontend/add-to-cart-variation.js

```diff
@@ -1,238 +1,218 @@
+		},
 
-					if ( value.length === 0 ) {
-						all_attributes_chosen = false;
-					} else {
-						some_attributes_chosen = true;
-					}
+		/**
+		 * Looks for matching variations for current selected attributes.
+		 */
+		onFindVariation: function( event ) {
+			var form = event.data.variationForm;
 
-					data[ attribute_name ] = value;
-				});
+			if ( form.useAjax ) {
+				return;
+			}
 
-				if ( all_attributes_chosen ) {
-					// Get a matchihng variation via ajax
-					data.product_id  = $product_id;
-					data.custom_data = $form.data( 'custom_data' );
+			form.$form.trigger( 'update_variation_values' );
 
-					$form.block( {
-						message: null,
-						overlayCSS: {
-							background: '#fff',
-							opacity: 0.6
-						}
-					} );
+			var attributes          = form.getChosenAttributes(),
+				matching_variations = form.findMatchingVariations( form.variationData, attributes.data );
 
-					$xhr = $.ajax( {
-						url: wc_cart_fragments_params.wc_ajax_url.toString().replace( '%%endpoint%%', 'get_variation' ),
-						type: 'POST',
-						data: data,
-						success: function( variation ) {
-							if ( variation ) {
-								$form.trigger( 'found_variation', [ variation ] );
-							} else {
-								$form.trigger( 'reset_data' );
-								$form.find( '.single_variation' ).after( '<p class="wc-no-matching-variations woocommerce-info">' + wc_add_to_cart_variation_params.i18n_no_matching_variations_text + '</p>' );
-								$form.find( '.wc-no-matching-variations' ).slideDown( 200 );
-							}
-						},
-						complete: function() {
-							$form.unblock();
-						}
-					} );
-				} else {
-					$form.trigger( 'reset_data' );
-				}
-				if ( some_attributes_chosen ) {
-					if ( $reset_variations.css( 'visibility' ) === 'hidden' ) {
-						$reset_variations.css( 'visibility', 'visible' ).hide().fadeIn();
-					}
+			if ( attributes.count === attributes.chosenCount ) {
+				var variation = matching_variations.shift();
+
+				if ( variation ) {
+					form.$form.trigger( 'found_variation', [ variation ] );
 				} else {
-					$reset_variations.css( 'visibility', 'hidden' );
+					window.alert( wc_add_to_cart_variation_params.i18n_no_matching_variations_text );
+					form.$form.trigger( 'reset_data' );
 				}
 			} else {
-				$form.trigger( 'woocommerce_variation_select_change' );
-				$form.trigger( 'check_variations', [ '', false ] );
-				$( this ).blur();
+				form.$form.trigger( 'reset_data' );
+				form.$singleVariation.slideUp( 200 ).trigger( 'hide_variation' );
 			}
 
-			// added to get around variation image flicker issue
-			$( '.product.has-default-attributes > .images' ).fadeTo( 200, 1 );
-
-			// Custom event for when variation selection has been changed
-			$form.trigger( 'woocommerce_variation_has_changed' );
-		} )
-
-		// Upon gaining focus
-		.on( 'focusin touchstart', '.variations select', function() {
-			if ( 'ontouchstart' in window || navigator.maxTouchPoints ) {
```

Comment:
Would anything break by not triggering `woocommerce_variation_select_focusin`? What are possible uses for this trigger (can't think of any after these changes)?